File: Data_Structures/3_Stack/C/main.c
#include <stdio.h>
#include <stdbool.h>
#include <stdlib.h>
#include "stack.h"

int main() {

    // Criando uma pilha
    Pilha* p = inicializaPilha();

    // Verificando se a pilha está vazia
    printf("A pilha está vazia? %s\n", estaVazia(p) ? "Sim" : "Não");

    // Adicionando elementos na pilha
    empilha(p, 10);
    printf("Elemento do topo da pilha: %d\n", topo(p));

    // Verificando tamanho da pilha
    printf("Tamanho da pilha: %d\n", tamanhoPilha(p));

    // Adicionando mais elementos na pilha
    empilha(p, 20);
    printf("Elemento do topo da pilha: %d\n", topo(p));

    // Removendo o elemento do topo da pilha
    printf("Elemento removido do topo da pilha: %d\n", desempilha(p));
    printf("Elemento do topo da pilha: %d\n", topo(p));

    return 0;
}

File: Data_Structures/3_Stack/Python/main.py
from stack import Stack

# Criando uma pilha 
pilha = Stack()

# Verificando se a pilha está vazia
print("A pilha está vazia? ", "Sim" if pilha.estaVazia() else "Não")

# Adicionando elementos na pilha
pilha.empilha(10)
print("Elemento do topo da pilha: ", pilha.topoElemento())

# Verificando tamanho da pilha
print("Tamanho da pilha: ", pilha.tamanho())

# Adicionando mais elementos na pilha
pilha.empilha(20)
print("Elemento do topo da pilha: ", pilha.topoElemento())

# Removendo o elemento do topo da pilha
print("Elemento removido do topo da pilha: ", pilha.desempilha())
print("Elemento do topo da pilha: ", pilha.topoElemento())
